rfd/0144-client-tools-updates.md

```diff
@@ -307,3 +307,148 @@ download server can use self-managed updates with system package managers which
 are signed.
 
 Phase 2 will use The Upgrade Framework (TUF) to implement secure updates.
+
+## Teleport Connect automatic updates *(added on 2025-06-27 by @gzdunek)*
+
+Teleport Connect is built with Electron and therefore cannot use the CLI update 
+mechanisms, which simply re-executes a command with a different tool version. 
+Instead, automatic updates will be implemented with the `electron-updater` package. 
+This library is maintained by the author of `electron-builder`, so it should be 
+fully compatible with our existing build and packaging setup. It handles all 
+the heavy lifting: downloading updates, verifying checksums, and installing 
+the updates.
+
+The library supports updates on macOS, Windows, and Linux, except the .tar.gz 
+target. Our initial release will not support updating .tar.gz builds; we may 
+explore this in the future.
+
+>Note: Electron's auto-update mechanism on macOS requires the ZIP target. 
+>It has already been added and will be available via direct download only - 
+>it will not be shown on the downloads page.
+
+### Custom update flow
+
+By default, `electron-updater` queries a static update server endpoint (like 
+GitHub) which returns a file containing the latest version metadata. 
+However, in our client tools update architecture, the latest version is dynamic, 
+based on `autoupdate_version` provided by clusters.
+
+To support this, a custom update provider will be implemented which will generate 
+the update metadata (like a download URL) based on the client tools version.
+
+### User experience
+
+Updates will be visible in two places:
+
+1. Login Dialog
+An auto-update widget will appear when the user opens the login dialog, 
+notifying them of a new version and prompting for a restart.
+It will be possible to open a detailed view from this widget.
+2. Additional Actions Menu
+This opens a detailed view containing a link to release notes, information about 
+cluster managing updates, and actions like canceling a download or re-checking 
+for updates.
```

Comment:
Thoughts on also adding this under Teleport Connect -> Check for Updates… in the app menu on macOS? I don't think it's an officially recommended approach (Apple would probably perefer you use App Store for updates), but many of the dev tools I use have a menu item for that there.

On macOS, it can exist both in the app menu and the additional actions menu, just so that in the docs and elsewhere we don't have to say "You can find it under additional actions on Windows and Linux and somewhere else on macOS".